.github/workflows/source_aws.yml

```diff
@@ -98,20 +119,23 @@ jobs:
       - name: Checkout
         if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
         uses: actions/checkout@v3
-      - uses: actions/cache@v3
-        if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
-        with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-1.19.5-release-cache-${{ hashFiles('plugins/source/aws/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-1.19.5-release-cache-plugins-source-aws
       - name: Set up Go
-        if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
+        id:   set-up-go
         uses: actions/setup-go@v3
         with:
-          go-version-file: plugins/source/aws/go.mod
+          go-version-file:       plugins/source/aws/go.mod
+          cache:                 true
+          cache-dependency-path: plugins/source/aws/go.sum
+      - name: Restore build cache
+        uses: actions/cache/restore@v3
+        with:
+          path:         |
+                        ~/.cache/go-build
+                        ~/Library/Caches/go-build
+                        ~\AppData\Local\go-build
+          key:          source-aws-${{ runner.os }}-go-${{ steps.set-up-go.outputs.go-version }}-${{ hashFiles('plugins/source/aws/go.sum') }}
```

Comment:
Let's also descope this change from the PR. Since this impacts release PRs and releases I'd rather do it separately (it changes caching so we use the same one for releases and test jobs, those run in parallel so I think we need to ensure there aren't race conditions here).
Also it seems the reason for this PR is to cache tests, which is unrelated to this change

Reply:
We want the build & test cache to be consistent. This change addresses this also by using the proper go version in the key file (see usage of setup-go outputs).
Additionally, there won't be any race conditions as the only place where we do save the cache is after push to `main` & testing (so that it has the most useful info for the future).

Reply:
> We want the build & test cache to be consistent. This change addresses this also by using the proper go version in the key file (see usage of setup-go outputs).

But it changes the cache key so the validate release and test workflows have the same key.
The reason for using this specific key in the validate release workflow is because it's used in the release workflow:
https://github.com/cloudquery/cloudquery/blob/33ebc306af2303174edc8bdc1a36e85cbcf88862/.github/workflows/release_plugin.yml#L44
https://github.com/cloudquery/cloudquery/blob/1621cf770f3c779335e2e6d38ab8e0d69ac92688/.github/workflows/source_aws.yml#L107

> so that it has the most useful info for the future

I'm not sure that's correct for releases. In releases the value is caching Go Releaser dry run, so next time we run it (the actual publish run) it's much faster, see example here:
https://github.com/cloudquery/cloudquery/actions/runs/4541185612/jobs/8003037262
![image](https://user-images.githubusercontent.com/26760571/229813611-2c08349b-6c54-46e1-a8b6-c9f23fa47e25.png)

Reply:
Regardless I don't think it's the purpose of this PR so we should remove this change from it and focus the PR on caching tests